### PR TITLE
(feat) UI improvements to the active visits widget

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
This PR introduces UI improvements to the active visits widget, which is currently present on the Home screen.

## Screenshots

Skeleton:
image

Error:
image

Empty state:
image

No results on searching
image

## Related Issue

Related to [O3-1878](https://issues.openmrs.org/browse/O3-1878?filter=-1)

## Other

*None.*

### CODE REVIEW

**Review by RandilaP — COMMENTED** [review]
There seems to be an issue with the end-to-end tests related to the visit summary page. I noticed that there's a timeout error occurring because of it. Would you be able to take a look at it and see if you can fix it?

**Comment by vasharma05** [comment]
Sure! Looking into it @RandilaP !

File: packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx [hunk 1]
```diff
@@ -168,116 +188,114 @@ const ActiveVisitsTable = () => {
...
-                }}
-              />
-            </TableContainer>
-          )}
-        </DataTable>
       </div>
-    );
-  }
+      <Search
+        // tabIndex={getBatchActionProps().shouldShowBatchActions ? -1 : 0}
```
Comment (hadijahkyampeire) [inline packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx:202]: could we just remove this comment to clean up the code?

File: packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx [hunk 2]
```diff
@@ -107,50 +104,73 @@ const ActiveVisitsTable = () => {
...
-  const {
-    goTo,
-    results: paginatedActiveVisits,
-    currentPage,
-  }: PaginationData = usePagination(searchResults, currentPageSize);
+    return activeVisits;
+  }, [searchString, activeVisits]);
 
-  const handleSearch = useCallback((e) => setSearchString(e.target.value), []);
+  const { paginated, goTo, results, totalPages, currentPage } = usePagination(searchResults, pageSize);
```
Comment (hadijahkyampeire) [inline packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx:123]: The `totalPages` is not used anywhere, we could just remove it.

**Review by hadijahkyampeire — APPROVED** [review]
Thanks @vasharma05

File: packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx (hunk 1 above)
Comment (vasharma05) [reply packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx:202]: Sure!
Thanks!

File: packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx (hunk 2 above)
Comment (vasharma05) [reply packages/esm-active-visits-app/src/active-visits-widget/active-visits.component.tsx:123]: Sure!
Thanks!

**Pull request merged**